### ISSUE TITLE
[BUG] Hyprlock crash in wlogout on VM

### ISSUE THREAD

**Issue by nanolookc** [mylinuxforwork/dotfiles#397]
**Describe the bug**
Hyprlock crashing if i turn off my monitor - when it started by wlogout. It's very strange, because if i start hyprlock via terminal, with the same command (~/.config/hypr/scripts/power.sh lock, or just hyprlock) - everything is working okay.

The only thing that is different from others - is that it's VM with gpu passthrough. 

**To Reproduce**
1. Click lock in the wlogout
2. Turn off monitor
3. Turn on monitor
4. See message about died hyprlock

**Expected behavior**
Well, working hyprlock

**Distribution (please complete the following information):**
 - EndeavourOS, proxmox VM

I also tried to access logs, and added to sh script this - 
if [[ "$1" == "lock" ]]; then
    echo ":: Lock"
    echo "Running hyprlock..." >> /home/dima/test.log
    sleep 0.5
    hyprlock > /home/dima/test.log
fi

but log is saved only then i launch log via terminal. File is empty when i start it via wlogout. 

Thanks for the project!

**Comment by nanolookc** [comment]
In recent checks I found out what it doesn't crash if wlogout launched from hotkey or terminal. It crashes only if wlogout was open from waybar button.

**Comment by mylinuxforwork** [comment]
> In recent checks I found out what it doesn't crash if wlogout launched from hotkey or terminal. It crashes only if wlogout was open from waybar button.

This is really strange. Shouldn’t be a different if starting from waybar or from the terminal.

**Comment by nanolookc** [comment]
If waybar launched from terminal - also same problem. Here is output (end is there i relaunch hyprlock from other tty)

❯ ~/.config/waybar/launch.sh
:: Theme: /ml4w-blur-custom

~
❯ [2024-11-04 16:36:38.138] [info] Using configuration file /home/dima/.config/waybar/themes/ml4w-blur-custom/config
[2024-11-04 16:36:38.138] [info] Including resource file: ~/.config/ml4w/settings/waybar-quicklinks.json
[2024-11-04 16:36:38.138] [info] Including resource file: ~/.config/waybar/modules.json
[2024-11-04 16:36:38.141] [info] Discovered appearance 'dark'
[2024-11-04 16:36:38.141] [info] Using CSS file /home/dima/.config/waybar/themes/ml4w-blur-custom/white/style.css
[2024-11-04 16:36:38.147] [info] Hyprland IPC starting
[2024-11-04 16:36:38.149] [info] Loading persistent workspaces from Waybar config
[2024-11-04 16:36:38.149] [info] Loading persistent workspaces from Hyprland workspace rules

~
❯ [2024-11-04 16:36:38.396] [info] Bar configured (width: 2880, height: 43) for output: DP-1
:: Lock
:: Lock
[2024-11-04 16:37:01.309] [info] Bar configured (width: 2880, height: 43) for output: DP-1

(waybar:25506): Gdk-CRITICAL **: 16:37:01.313: gdk_monitor_get_scale_factor: assertion 'GDK_IS_MONITOR (monitor)' failed
[2024-11-04 16:37:01.335] [info] Bar removed from output: DP-1

(waybar:25506): Gdk-CRITICAL **: 16:37:01.335: gdk_monitor_get_scale_factor: assertion 'GDK_IS_MONITOR (monitor)' failed

(waybar:25506): Gdk-CRITICAL **: 16:37:01.335: gdk_monitor_get_scale_factor: assertion 'GDK_IS_MONITOR (monitor)' failed

(waybar:25506): Gdk-CRITICAL **: 16:37:01.335: gdk_monitor_get_scale_factor: assertion 'GDK_IS_MONITOR (monitor)' failed

(waybar:25506): Gdk-CRITICAL **: 16:37:01.335: gdk_monitor_get_scale_factor: assertion 'GDK_IS_MONITOR (monitor)' failed

(waybar:25506): Gdk-CRITICAL **: 16:37:01.335: gdk_monitor_get_scale_factor: assertion 'GDK_IS_MONITOR (monitor)' failed

(waybar:25506): Gdk-CRITICAL **: 16:37:01.335: gdk_monitor_get_scale_factor: assertion 'GDK_IS_MONITOR (monitor)' failed
[2024-11-04 16:37:07.229] [info] Loading persistent workspaces from Waybar config
[2024-11-04 16:37:07.229] [info] Loading persistent workspaces from Hyprland workspace rules
[2024-11-04 16:37:07.232] [warning] module backlight: Disabling module "backlight", No backlight found
[2024-11-04 16:37:07.234] [warning] No batteries.
[2024-11-04 16:37:07.365] [info] Bar configured (width: 2880, height: 43) for output: DP-1
ok
ok
ok
ok

**Comment by mylinuxforwork** [comment]
@nanolookc Are all packages up-to-date?

**Comment by nanolookc** [comment]
Yep, latest hyprland/hyprlock/waybar/configs. Still same with hyprland 45.

**Comment by mylinuxforwork** [comment]
@nanolookc Is it still open with Hyprland 0.48.x?
